addons/html_builder/static/src/core/builder_options_plugin.js

```diff
@@ -53,10 +55,19 @@ export class BuilderOptionsPlugin extends Plugin {
     };
 
     setup() {
-        this.builderOptions = withIds(this.getResource("builder_options"));
+        this.builderOptions = this.getResource("builder_options");
+        this.builderOptionsContext = new Map();
+        this.builderOptionsDependencies = new Map();
+        const options = this.builderOptions.concat([OptionsContainer]);
+        for (const Option of options) {
+            this.getBuilderDependencies(Option);
+            this.getBuilderOptionContext(Option);
+        }
```

Comment:
Question: We agree that this is only needed because we set dependencies inside the options ? Like if we just exposed the whole editor to the options, we normally wouldn't need it ?